services/hanke-service/src/main/resources/db/changelog/changesets/086-change-car-lane-nuisance-enum.sql

```diff
@@ -14,41 +14,40 @@ UPDATE hankealue
         END;
 
 -- update applications.applicationData.areas
-WITH areas_array AS (
+WITH updated_areas AS (
     SELECT id,
         CASE
             WHEN (applicationData #>> '{areas}') IS NOT NULL AND jsonb_typeof(applicationData->'areas') = 'array' THEN
                 jsonb_set(
                     applicationData,
                     '{areas}',
-                    (
-                        SELECT jsonb_agg(
-                            jsonb_set(
-                                area,
-                                '{kaistahaitta}',
-                                (
+                    jsonb_agg(
+                        CASE
+                            WHEN (area #>> '{kaistahaitta}') IS NOT NULL THEN
+                                jsonb_set(
+                                    area,
+                                    '{kaistahaitta}',
                                     CASE area->>'kaistahaitta'
-                                        WHEN 'VAHENTAA_KAISTAN_YHDELLA_AJOSUUNNALLA' THEN
-                                            '"YKSI_KAISTA_VAHENEE"'::jsonb
-                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_KAISTAN_KAHDELLA_AJOSUUNNALLA' THEN
-                                            '"YKSI_KAISTA_VAHENEE_KAHDELLA_AJOSUUNNALLA"'::jsonb
-                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_USEITA_KAISTOJA_KAHDELLA_AJOSUUNNALLA' THEN
-                                            '"USEITA_KAISTOJA_VAHENEE_AJOSUUNNILLA"'::jsonb
-                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_USEITA_KAISTOJA_LIITTYMIEN_ERI_SUUNNILLA' THEN
-                                            '"USEITA_AJOSUUNTIA_POISTUU_KAYTOSTA"'::jsonb
+                                        WHEN 'VAHENTAA_KAISTAN_YHDELLA_AJOSUUNNALLA' THEN '"YKSI_KAISTA_VAHENEE"'::jsonb
+                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_KAISTAN_KAHDELLA_AJOSUUNNALLA' THEN '"YKSI_KAISTA_VAHENEE_KAHDELLA_AJOSUUNNALLA"'::jsonb
+                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_USEITA_KAISTOJA_KAHDELLA_AJOSUUNNALLA' THEN '"USEITA_KAISTOJA_VAHENEE_AJOSUUNNILLA"'::jsonb
+                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_USEITA_KAISTOJA_LIITTYMIEN_ERI_SUUNNILLA' THEN '"USEITA_AJOSUUNTIA_POISTUU_KAYTOSTA"'::jsonb
                                         ELSE area->'kaistahaitta'
                                     END
                                 )
-                            )
-                        )
-                        FROM jsonb_array_elements(applicationData->'areas') area
+                            ELSE area
+                        END
                     )
                 )
             ELSE applicationData
-        END as updated_data
-    FROM applications
+        END AS updated_data
+    FROM applications,
+         jsonb_array_elements(applicationData->'areas') AS area
```

Comment:
When I have a newly created cable report application (just created from the UI with no edits) in the DB I get an error:
```
ERROR:  cannot extract elements from a scalar
```
I think it's related to this line. The newly created cable report has a null `applicationdata.areas`. It's not even an empty list, just null. So this function can't run on it.

There could be a WHERE clause to limit this query to only the cases where the areas are not null. Which is getting pretty close to the suggestion I made earlier about moving the outermost CASE to a WHERE.

Reply:
I simplified the query to do the update within a single UPDATE clause.

Reply:
It would be nice if your answer had something to do with the comment above it.